src/Parsers/IAST.cpp

```diff
@@ -260,6 +263,21 @@ void IAST::FormatSettings::writeIdentifier(const String & name, bool ambiguous)
     }
 }
 
+void IAST::FormatSettings::checkIdentifier(const String & name) const
```

Comment:
1. `checkIdentifier` is a too generic and non-speaking name. Maybe `validateIdentifierIsSecure`?

Reply:
2. Someone reading the code wonders why we check at all if the identifier is secure. Did `executeQuery` not convert all identifiers to secure identifiers? If this is a sanity check that is not expected to trigger in production, please add a comment "sanity check" etc.

Reply:
> 1. checkIdentifier is a too generic and non-speaking name. Maybe validateIdentifierIsSecure?

In the future, there might be more validations for identifiers. We just put all of them into this function. That's why I just used a general name. I will check it to `validateIdentifier`. WDYT?

> 2. Someone reading the code wonders why we check at all if the identifier is secure. Did executeQuery not convert all identifiers to secure identifiers? If this is a sanity check that is not expected to trigger in production, please add a comment "sanity check" etc.

"secure identifiers" might be misleading. "restricted identifiers" might sound more appropriate. When reading the code, if the setting is enabled, it means that users want to restrict the identifier.

Reply:
`validateIdentifier` --> Ok